packages/pedantic/lib.js

```diff
@@ -0,0 +1,122 @@
+const path = require('path');
+const chalk = require('chalk');
+const { promises: fs } = require('fs');
+const { default: sortImports } = require('import-sort');
+const { debuglog } = require('util');
+const prettier = require('prettier');
+const { getConfig } = require('import-sort-config');
+
+const ConsoleUtilities = require('@4c/cli-core/ConsoleUtilities');
+const ArgUtilities = require('@4c/cli-core/ArgUtilities');
+
+const debug = debuglog('pedantic');
+
+function sortFileImports(content, filePath) {
+  const resolvedConfig = getConfig(
+    path.extname(filePath),
+    path.dirname(filePath),
+  );
+
+  if (!resolvedConfig || !resolvedConfig.parser || !resolvedConfig.style) {
+    debug('could not resolve import sort config for:', filePath);
+    return { code: content, changes: [] };
+  }
+
+  const { parser, style, options } = resolvedConfig;
+  const result = sortImports(content, parser, style, filePath, options);
+
+  return result.code;
+}
+
+async function runPrettier(content, filePath, ignorePath) {
+  const { ignored } = await prettier.getFileInfo(filePath, { ignorePath });
+  if (ignored) return content;
+  const options = await prettier.resolveConfig(filePath);
+
+  return prettier.format(content, { filepath: filePath, ...options });
+}
+
+module.exports = async (
+  filePatterns,
+  {
+    cwd = process.cwd(),
+    ignorePath,
+    ignoreNodeModules,
+    write,
+    check,
+    listDifferent,
+  },
+) => {
+  const different = [];
+  debug('patterns:', filePatterns, 'write:', write, 'cwd', cwd);
+  console.log('Checking formatting…');
+
+  const filePaths = await ArgUtilities.resolveFilePatterns(filePatterns, {
+    ignoreNodeModules,
+    cwd,
+  });
+
+  let spinner;
+  if (ConsoleUtilities.isTTY()) {
+    spinner = ConsoleUtilities.spinner();
+  }
+
+  if (!filePaths.length) {
+    process.exitCode = 1;
+    if (spinner) spinner.fail();
+    console.error(
+      "The provided file patterns didn't match any files: ",
+      filePatterns.join(', '),
+    );
+  }
+
+  await Promise.all(
+    filePaths.map(async filePath => {
+      let content;
+      let code;
+
+      try {
+        content = await fs.readFile(filePath, 'utf8');
+
+        code = sortFileImports(content, filePath);
+        code = await runPrettier(code, filePath, ignorePath);
+      } catch (err) {
```

Comment:
does it make sense to run eslint here as well?

Reply:
i'd like to do that, i'm not sure i can make eslint work as a pure formatter tho. It's annoying that it exits loudly for non fixable problems, but maybe avoidable via the node api

Reply:
in practice that's probably good enough for it to fail loudly on other stuff, as long as we run it after e.g. import-sort/&c. (which we don't right now)